src/content/doc-sdk-javascript/frameworks/solidjs.mdx

```diff
@@ -0,0 +1,397 @@
+							... your routes here
+						</Route>
+					</Router>
+				</AuthProvider>
+			</SurrealProvider>
+		</QueryClientProvider>
+	);
+};
+
+export default App;
+```
+
+### 4. Use the `useSurreal` Hook in Your Components
+
+Now you can use the `useSurreal` hook in your components to access the SurrealDB client. For example, you can create an `AuthProvider` to handle authentication and user management.
+
+```ts title="src/lib/providers/AuthProvider.tsx"
+import { Accessor, createContext, ParentProps, useContext } from "solid-js";
+import { createStore } from "solid-js/store";
+import { useSurreal } from "../lib/providers/SurrealProvider";
+
+interface AuthProviderState {
+	user: Accessor<UserRecord | undefined>;
+	login: (email: string, password: string) => Promise<void>;
+	register: (data: Omit<UserRecord, "id">) => Promise<void>;
+	logout: () => Promise<void>;
+}
+
+interface AuthProviderStore {
+	user: UserRecord | undefined;
+	status: "idle" | "signing-in" | "signing-up" | "signing-out" | "signed-in" | "signed-up";
+}
+
+const AuthContext = createContext<AuthProviderState>();
+
+export function AuthProvider(props: ParentProps) {
+
+	const { 
+		client, 
+		close, 
+		connect 
+	} = useSurreal();
+
+	const [store, setStore] = createStore<AuthProviderStore>({
+		user: undefined,
+		status: "idle"
+	});
+
+	const login = async (email: string, password: string) => {
+		const db = client();
+		
+		setStore("status", "signing-in");
+
+		await db.signin({ 
+			access: "user",
+			namespace: "surrealdb",
+			database: "pollwebapp",
+			variables: {
+				email: email,
+				pass: password
+			}
+		});
+
+		setStore("status", "signed-up");
+	};
+
+	const register = async (data: Omit<UserRecord, "id">) => {
+		const db = client();
+
+		setStore("status", "signing-up");
+
+		await db.signup({
+			access: "user",
+			namespace: "surrealdb",
+			database: "pollwebapp",
+			variables: data
+		}); 
+
+		setStore("status", "signed-up");
+	};
+
+	const logout = async () => {
+
+		setStore("status", "signing-out");
+
+		await close();
+		await connect();
+
+		setStore("status", "idle");
+	};
+
+	const providerValue: AuthProviderState = {
+		user: () => store.user,
+		login,
+		register,
+		logout
+	};
+
+	return (
+		<AuthContext.Provider value={providerValue}>
+			{props.children}
+		</AuthContext.Provider>
+	);
+}
+
+export function useAuth(): AuthProviderState {
+	const ctx = useContext(AuthContext);
+
+	if(!ctx) {
+		throw new Error("useAuth must be used within an AuthProvider");
+	}
+
+	return ctx;
+}
+```
+
+## Next steps 
+
+Now that you have SurrealDB integrated into your SolidJS project, you're ready to start building! To learn more about interacting with your database:
+
+- Explore the [JavaScript SDK reference](/docs/sdk/javascript/methods) for a complete list of available methods and features
+- See SurrealDB in action by checking out our SolidJS example projects in the [examples section](/docs/sdk/javascript#example-projects)
```

Comment:
question: there is no SolidJS example AFAIK, am I right?

Reply:
Not yet but we're literally working on one right now 😁

Reply:
https://github.com/welpie21/hackathon-poll-webapp